redis_json/src/c_api.rs

```diff
@@ -45,6 +45,11 @@ struct ResultsIterator<'a, V: SelectValue> {
     pos: usize,
 }
 
+struct KeyValuesIterator<'a, V: SelectValue> {
+    results: Vec<(&'a str, &'a V)>,
```

Comment:
Can we keep the iterator over the iterms here instead of collecting it into a vector?

Reply:
I tried and it crashed and I wanted to look at it later due to other priorities
https://github.com/RedisJSON/RedisJSON/pull/1047/commits/61570c41f13fc7c330103c74a9fd8a1f678a1e05

Reply:
So let understand why it crashes

Reply:
Was missing a `Box::new` in `json_api_get_key_value`